apps/sparo-lib/src/logic/GitVersionCompatibility.ts

```diff
@@ -1,25 +1,33 @@
 import 'reflect-metadata';
-import { getFromContainerAsync } from '../di/container';
+import { getFromContainer } from '../di/container';
 import { GitService } from '../services/GitService';
 
 /**
  * This class provides the useful function to check git version.
  */
 export class GitVersionCompatibility {
-  private constructor() {}
+  private static _gitVersion: [number, number, number] | undefined;
 
-  public static async ensureGitVersionAsync(): Promise<void> {
-    const gitService: GitService = await getFromContainerAsync(GitService);
-    const gitVersion: [number, number, number] | undefined = gitService.getGitVersion();
-    if (!gitVersion) {
-      throw new Error(`Fail to get git version`);
-    }
+  private constructor() {}
 
-    const [major, minor, patch] = gitVersion;
+  public static ensureGitVersion(): void {
+    const [major, minor, patch] = GitVersionCompatibility.getGitVersion();
     if (major < 2 || minor < 32) {
       throw new Error(
         `git version is too low. The minimal git version is >=2.32.0. Your git version is ${major}.${minor}.${patch}. Please upgrade git.`
       );
     }
   }
+
+  public static getGitVersion(): [number, number, number] {
+    if (!GitVersionCompatibility._gitVersion) {
+      const gitService: GitService = getFromContainer(GitService);
+      const gitVersion: [number, number, number] | undefined = gitService.getGitVersion();
+      if (!gitVersion) {
+        throw new Error(`Fail to get git version`);
```

Comment:
```suggestion
        throw new Error(`Fail to detect Git version`);
```
The "version" is more naturally considered to be the version of the "Git" software, not just the `git` binary.